src/flows/context/query.tsx

```diff
@@ -112,83 +110,133 @@ export const QueryProvider: FC = ({children}) => {
     }
   }, [bucketsLoadingState, dispatch])
 
-  const vars = useMemo(() => {
-    if (flow?.range) {
-      return variables
+  const query = (text: string, vars: VariableMap = {}): Promise<FluxResult> => {
+    // Some preamble for setting the stage
+    const baseAST = parse(text)
+    const usedVars = _getVars(baseAST, vars)
+    const optionAST = buildVarsOption(
+      Object.values(usedVars)
+        .filter(v => !!v)
         .map(v => asAssignment(v))
-        .concat(getTimeRangeVars(flow.range))
-    }
-
-    variables.map(v => asAssignment(v))
-  }, [variables, flow?.range])
-
-  const generateMap = (withSideEffects?: boolean): Stage[] => {
-    return flow.data.allIDs
-      .reduce((stages, pipeID) => {
-        const pipe = flow.data.get(pipeID)
-
-        const stage = {
-          text: '',
-          instances: [pipeID],
-          requirements: {},
-        }
-
-        const create = text => {
-          if (text && PREVIOUS_REGEXP.test(text) && stages.length) {
-            stage.text = text.replace(
-              PREVIOUS_REGEXP,
-              stages[stages.length - 1].text
-            )
-          } else {
-            stage.text = text
-          }
+    )
 
-          stages.push(stage)
-        }
+    // Make a version of the query with the variables loaded
+    const ast = {
+      package: '',
+      type: 'Package',
+      files: [baseAST, optionAST],
+    }
 
-        const append = () => {
-          if (stages.length) {
-            stages[stages.length - 1].instances.push(pipeID)
+    // Here we grab the org from the contents of the query
```

Comment:
could some of this be pulled out into a couple functions to make this more readable? Like the stuff between lines 130 and 236.

Reply:
sure

Reply:
hope that did it